src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1608,6 +1608,25 @@ public ScrollListener(int dir, boolean block)   {
 
         /** {@inheritDoc} */
         public void actionPerformed(ActionEvent e) {
+            // If frame is disabled and time is started in mousePressed
+            // and mouseReleased is not called, then timer will not be stopped
+            // Stop the timer if frame is disabled
+            Component parent = scrollbar.getParent();
+            do {
+                if (parent instanceof javax.swing.JFrame par) {
+                    if (!par.isEnabled()) {
```

Comment:
I think this two condition can be combined into one.

`if (parent instanceof javax.swing.JFrame par && !par.isEnabled())
`

Reply:
No, we need to break if frame is enabled and not look for its parent..

Reply:
ok.